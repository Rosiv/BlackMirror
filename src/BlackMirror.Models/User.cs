namespace BlackMirror.Models
{
    using System.Collections.Generic;
    using BlackMirror.Interfaces;
    public class User : IUser
    {
        public string Id { get; set; }

        public string Name { get; set; }

        public string Email { get; set; }

        public IEnumerable<ICredentials> RepositoryCredentials { get; set; }
    }
}