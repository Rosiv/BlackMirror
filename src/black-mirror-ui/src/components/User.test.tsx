import * as React from 'react';
import renderer from 'react-test-renderer';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
const customMuiTheme = getMuiTheme(darkBaseTheme);
import User from './User';

it('renders and is compatible with the snapshot', () => {
    const tree = renderer
        .create(
        <MuiThemeProvider muiTheme={customMuiTheme}>
            <User
                user={'Pro'}
                nestedLevel={0}
                secondaryText={'Admin'}
            />
        </MuiThemeProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
