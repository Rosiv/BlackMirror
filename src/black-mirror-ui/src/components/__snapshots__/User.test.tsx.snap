// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and is compatible with the snapshot 1`] = `
<div>
  <span
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "backgroundColor": null,
        "border": 10,
        "boxSizing": "border-box",
        "color": "rgba(255, 255, 255, 1)",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "fontWeight": "inherit",
        "lineHeight": "16px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
      }
    }
    tabIndex={0}
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "marginLeft": 0,
            "muiPrepared": true,
            "paddingBottom": 16,
            "paddingLeft": 72,
            "paddingRight": 16,
            "paddingTop": 20,
            "position": "relative",
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "display": "block",
              "fill": "#00bcd4",
              "height": 24,
              "left": 4,
              "margin": 12,
              "muiPrepared": true,
              "position": "absolute",
              "top": 12,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          />
        </svg>
        <div
          style={
            Object {
              "muiPrepared": true,
            }
          }
        >
          Pro
        </div>
        <div
          style={
            Object {
              "WebkitBoxOrient": null,
              "WebkitLineClamp": null,
              "color": "rgba(255, 255, 255, 0.7)",
              "display": null,
              "fontSize": 14,
              "height": 16,
              "lineHeight": "16px",
              "margin": 0,
              "marginTop": 4,
              "muiPrepared": true,
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          Admin
        </div>
      </div>
    </div>
  </span>
</div>
`;
